src/screens/Schedule/ScheduleScreen.tsx

```diff
@@ -38,20 +43,28 @@ const ScheduleScreen = () => {
         setCalendlyLinks(filteredLinks);
 
         // fetch all uncancelled appointments for client
-        const appts = await getAllAppointmentsForClient(client);
-        // TO DO: filter out past appointments
+        const appts = await getAllUpcomingAppointmentsForClient(client);
         setAppointments(appts);
       }
     }
     loadLinksAndAppointments();
-  }, []);
+  }, [isFocused, detectBrowserClose]);
 
   const openCalendlyInBrowser = async (link: string) => {
-    await WebBrowser.openBrowserAsync(link);
+    if (Platform.OS === 'ios') {
+      await WebBrowser.openBrowserAsync(link);
+      setDetectBrowserClose(!detectBrowserClose);
+      console.log('ios');
+    } else {
+      await WebBrowser.openAuthSessionAsync(link, '');
```

Comment:
Is there a reason for handling browser differently for Android vs iOS? This seems to work fine for iOS, but when I try opening the browser in Android emulator, I get this error. 
<img width="320" alt="Screen Shot 2021-11-21 at 5 42 12 PM" src="https://user-images.githubusercontent.com/63089231/142788898-821e270f-8431-44e9-9bc9-61963efc252d.png">
ee

Reply:
`openBrowserAsync` behaves differently on ios and android. for ios, it behaves as desired, returning from the function call when the browser is closed. on android, it returns when the browser is opened, so we need to use a different function to detect when the web browser is closed.

Reply:
hmm it runs correctly for me - did you npm install?